test_macros/src/lib.rs

```diff
@@ -24,12 +24,25 @@ impl Parse for TestInput {
 #[proc_macro]
 pub fn ctest(input: TokenStream) -> TokenStream {
     let input = parse_macro_input!(input as TestInput);
-    impl_ciphersuite_tests(input).unwrap()
+    impl_ciphersuite_tests(input, false).unwrap()
 }
 
-fn impl_ciphersuite_tests(input: TestInput) -> Result<TokenStream> {
+#[proc_macro]
+pub fn ctest_panic(input: TokenStream) -> TokenStream {
+    let input = parse_macro_input!(input as TestInput);
+    impl_ciphersuite_tests(input, true).unwrap()
+}
+
+fn impl_ciphersuite_tests(input: TestInput, should_panic: bool) -> Result<TokenStream> {
```

Comment:
You can pass in a `proc_macro::TokenStream` here and the `quote`s as arguments to safe you the conditional.